src/main/fc/tasks.c

```diff
@@ -383,7 +383,7 @@ task_attribute_t task_attributes[TASK_COUNT] = {
 #endif
 
 #ifdef USE_MAG
-    [TASK_COMPASS] = DEFINE_TASK("COMPASS", NULL, NULL, taskUpdateMag, TASK_PERIOD_HZ(10), TASK_PRIORITY_LOW),
+    [TASK_COMPASS] = DEFINE_TASK("COMPASS", NULL, NULL, taskUpdateMag, TASK_PERIOD_HZ(200), TASK_PRIORITY_LOW),
```

Comment:
I'd try 100Hz first. I2C is quite slow, at 200Hz, it is about 10% of bus bandwidth..

Reply:
Thanks.
I'll test 100Hz and report back.
There will be occasional spikes since mag reads and IMU are asynchronous. We are not using derivative of heading, so these should not be a big issue.